### PR TITLE
approxmc: init at 4.1.23

### LINKED ISSUE #275864: Package request: approxmc
**Project description**

`approxmc` is an approximate model counter for CNF formulas.

**Metadata**

* homepage URL: https://github.com/meelgroup/approxmc
* source URL: https://github.com/meelgroup/approxmc
* license: mit
* platforms: linux

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by t4ccer**
## Description of changes

Closes https://github.com/NixOS/nixpkgs/issues/275864

Added required dependencies as well

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ap/approxmc/package.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    louvain-community
+  ];
+
+  meta = with lib; {
+    description = "Approximate Model Counter";
+    homepage = "https://github.com/meelgroup/approxmc";
+    license = licenses.mit;
+    maintainers = with maintainers; [ t4ccer ];
+    mainProgram = "approxmc";
+  };
```
Comment (AndersonTorres) [inline pkgs/by-name/ap/approxmc/package.nix:41]: Set meta.platforms

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ar/arjun-cnf/package.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    hash = "sha256-5duc05s654HLjbf+dPgyMn6QUVvB0vLji3M4S2o/QYU=";
+  };
+
+  # Can be removed after next release
+  patches = [
+    (fetchpatch {
+      url = "https://github.com/meelgroup/arjun/commit/34188760f1ab4b1b557c45ccaee8d2b9b6f0b901.patch";
+      hash = "sha256-E/yk2ohHP2BAFg353r8EU01bZCqeEjvpJCrBsxPiOWM=";
+    })
+    # Based on https://github.com/meelgroup/arjun/commit/99c4ed4ad820674632c5d9bbcc98c001f8cac98f
```
Comment (kirillrdy) [inline pkgs/by-name/ar/arjun-cnf/package.nix:28]: whats is the motivation for not using that patch as is ?

File: pkgs/by-name/lo/louvain-community/package.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  src = fetchFromGitHub {
+    owner = "meelgroup";
+    repo = "louvain-community";
+    rev = "8cc5382d4844af127b1c1257373740d7e6b76f1e";
+    hash = "sha256-0i3wrDdOyleOPv5iVO1YzPfTPnIdljLabCvl3SYEQOs=";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  meta = with lib; {
```
Comment (kirillrdy) [inline pkgs/by-name/lo/louvain-community/package.nix:20]: set `platforms` here as well

File: pkgs/by-name/ar/arjun-cnf/package.nix (hunk 2 above)
Comment (t4ccer) [reply pkgs/by-name/ar/arjun-cnf/package.nix:28]: Upstream patch has a conflict and cannot be applied

**Comment by t4ccer** [comment]
Added `meta.platforms`

**Review by AndersonTorres — APPROVED** [review]

**Comment by emeinhardt** [comment]
Since the package request was only for linux and the failing builds are only for `*-darwin`, can the release just include `linux` for the time being?

**Comment by t4ccer** [comment]
Ehh it should build on darwin but I don't feel like catching type missmatch without access to darwin machine. Changed platforms to `linux` on things that were not building

**Comment by kirillrdy** [comment]
@t4ccer thank you. Darwin support can be done in subsequent PR feel free to ping me

**Pull request merged**